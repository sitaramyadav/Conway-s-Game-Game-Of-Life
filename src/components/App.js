import React, { useState } from "react";
import styled from "styled-components";
import { Board } from "./Board";
const Header = styled.header`
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  width: 100%;
  height: 70px;
  background-color: black;
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;
`;
const GameStatus = styled.button`
  width: 90px;
  background-color: black;
  color: white;
`;

function App() {
  const [gameStatus, setGameStatus] = useState(false);
  return (
    <div>
      <Header>Game of life</Header>
      <Board />
      <GameStatus
        onClick={() => setGameStatus(!gameStatus)}
        gameStatus={gameStatus}
      />
    </div>
  );
}

export default App;
