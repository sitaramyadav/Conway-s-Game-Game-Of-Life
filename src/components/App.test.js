import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
// import "@testing-library/jest-dom/extend-expect";

describe("App", () => {
  it("should render the child components", () => {
    const { queryByTestId } = render(<App />);

    expect(queryByTestId("App")).toBeInTheDocument();
  });
});
