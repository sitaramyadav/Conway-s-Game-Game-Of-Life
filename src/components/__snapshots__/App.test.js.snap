// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the child components 1`] = `[Function]`;
